src/parser/parse_tree_cl.c

```diff
@@ -5209,13 +5209,15 @@ pt_init_print_f (void)
 void
 pt_init_node (PT_NODE * node, PT_NODE_TYPE node_type)
 {
-  PARSER_INIT_NODE_FUNC parser_init_node_func = pt_init_func_array[node_type];
-  if (!node || !parser_init_node_func)
+  assert (node_type < PT_LAST_NODE_NUMBER);
+  assert (pt_init_f != NULL);
```

Comment:
It seems no to be needed because pt_init_f is initialized at parser creating.
I wonder the you use **pt_init_f**  instead of **parser_init_node_func**

Reply:
In either case, if pt_init_f is non-null, the parser_init_node_func array will match the appropriate functions.
However, if pt_init_f is null, the parser_init_node_func array contains pointers to meaningless garbage values. That is, it has not been initialized.
So I think pt_init_f is more appropriate.
And there is only one place here that uses parser_init_node_func

Reply:
but by assert (pt_init_f != NULL) it is proved that pt_init_f is not null.